TaskWithSmartPointers/MySharedPtr.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+#include <iostream>
+
+template <class T>
+class MySharedPtr
+{
+private:
+	T* m_ptr = nullptr;
+	long* m_referencesCounter = nullptr;
```

Comment:
Please investigate if we can use std::atomic_long  as the reference counter (a moral equivalent of std::atomic<long>).

Reply:
If we can, then we don't need to guard variables with std::mutex.

Reply:
1) I left comments related to trying to use std::atomic, gives an error, decided not to stop there
2) I used the example of std::mutex, perhaps not quite in the correct places, but everything works